class.frame-nonce-preview.php

```diff
@@ -45,15 +52,19 @@ public function is_frame_nonce_valid() {
 			return false;
 		}
 
-		Jetpack::load_xml_rpc_client();
-		$xml = new Jetpack_IXR_Client();
-		$xml->query( 'jetpack.verifyFrameNonce', sanitize_key( $_GET['frame-nonce'] ) );
+		if ( empty( $this->frame_nonce_valid ) ) {
+			Jetpack::load_xml_rpc_client();
+			$xml = new Jetpack_IXR_Client();
+			$xml->query( 'jetpack.verifyFrameNonce', sanitize_key( $_GET['frame-nonce'] ) );
 
-		if ( $xml->isError() ) {
-			return false;
+			if ( $xml->isError() ) {
+				return false;
+			}
+
+			$this->frame_nonce_valid = (bool) $xml->getResponse();
```

Comment:
If this returns `false`, we'll keeping checking, since the `empty()` check above will evaluate to `true` if `frame_nonce_valid` is `false`. I can't decide if that's good or not 🙃

Reply:
yeah, we should probably be checking against `null`